### ISSUE TITLE
Not functioning with Wayland (Sway), Arch Linux package also needs fix

### ISSUE THREAD

**Issue by krompus** [foresto/joystickwake#10]
I tried installing joystickwake from the AUR: https://aur.archlinux.org/packages/joystickwake on my Arch Linux + Sway https://swaywm.org/ install, it installs without issue. I'm not sure why, but it just outputs `command not found` in the console afterwards. Afterwards, I cloned it directly from this repo and it runs, but does not function. Would appreciate some help getting this to work, thanks!

https://gist.github.com/krompus/7a6e4a85f0b72099b145a39a05899761#file-joystickwake-log

**Comment by foresto** [comment]
Joystickwake delegates screen waking to whatever command your system has installed to do that job. It can be one of the common commands that it already knows about, or a custom one that you configure yourself.

That log is telling you that, of all the screen-waking commands joystickwake knows to try by default, only `xset s reset` ran successfully on your system. (And that you configured `test command` as a custom one, which ran without error because `test` is a valid unix command that does nothing but return an exit code, but of course that won't wake the screen.)

To get it working, you must figure out what command will poke your screensaver into backing off, and configure joystickwake to run that command.

**Comment by foresto** [comment]
I don't know how screen blanking (and waking) is usually managed with Sway, but if you find out what command will do the trick in a widely-used environment, I would consider adding it to joystickwake's default list.

**Comment by goremykin** [comment]
for KDE Plasma Wayland use `qdbus-qt5 org.freedesktop.ScreenSaver /ScreenSaver SimulateUserActivity`

**Comment by foresto** [comment]
Closing, since there is no bug here.

However, I am still willing to add a default waker for Sway, if a widely used one is identified. If someone wants to point me at the exact command line for such a thing, and documentation for it, please don't hesitate to comment.

**Comment by foresto** [comment]
As far as I can tell, Sway's only way of suppressing the screen blanker is through Wayland's experimental [zwp_idle_inhibit_manager_v1](https://wayland.app/protocols/idle-inhibit-unstable-v1) protocol. That presents at least two problems:

- It is a stateful interface, requiring the calling process to run for as long as the screen is to be kept awake. It doesn't support the heartbeat-style commands used by joystickwake at all. Implementing it would demand much more of my time than a simple heartbeat command, and add desktop-specific complexity to joystickwake, which I have tried to avoid so far.
- There doesn't seem to be a suitable Wayland protocol implementation for Python yet. Of the two I found, one is unmaintained and undocumented, and the other is neither pure Python nor widely available in linux distros.

I won't entirely rule it out based on that first point, especially since it looks like swayidle might provide heartbeat-style waking through the [org_kde_kwin_idle](https://wayland.app/protocols/kde-idle#org_kde_kwin_idle_timeout:request:simulate_user_activity) interface.

However, that second point is a blocker. I'm not likely to build a Wayland library from scratch, so until a suitable one emerges, considering this will have to wait.

Since you're one of their users, you might ask the Sway developers if they plan to implement the org.freedesktop.ScreenSaver D-Bus interface. Ideally with the SimulateUserActivity method that KDE offers. That's something I could see joystickwake supporting fairly soon.
